### PR TITLE
Clean up package dependencies

### PR DESCRIPTION
**Pull request by reyang**
Please refer to my PR comments for the context.

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -7,16 +7,17 @@
     <OTelLatestStableVer>1.5.1</OTelLatestStableVer>
   </PropertyGroup>
 
-  <!-- prod packages -->
+  <!--
```
Comment (reyang) [inline Directory.Packages.props:10]: I found that "prod packages" have never been clarified and the current state is messy.
So here I added clarification, also kicked some packages from the "prod" section to the "nonprod" section.

File: build/Common.props [hunk 2]
```diff
@@ -27,13 +27,11 @@
...
-    <PackageReference Include="StyleCop.Analyzers" Condition="'$(SkipAnalysis)'!='true'">
-      <PrivateAssets>All</PrivateAssets>
-    </PackageReference>
+    <PackageReference Include="StyleCop.Analyzers" PrivateAssets="All" Condition="'$(SkipAnalysis)'!='true'" />
     <PackageReference Include="Microsoft.CodeCoverage" PrivateAssets="All" Condition="'$(Configuration)'=='Release'" />
-    <!--<PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Condition="'$(SkipAnalysis)'!='true'">
-      <PrivateAssets>all</PrivateAssets>
-    </PackageReference>-->
+    <!--
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" PrivateAssets="All" Condition="'$(SkipAnalysis)'!='true'" />
```
Comment (reyang) [inline build/Common.props:33]: This should be an editorial change, note that previously we were using "All" and "all", inline vs. property without consistency (which makes it hard to search).

File: test/OpenTelemetry.Api.Tests/OpenTelemetry.Api.Tests.csproj [hunk 3]
```diff
@@ -21,11 +21,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPkgVer)" />
-    <PackageReference Include="Moq" Version="$(MoqPkgVer)" />
```
Comment (reyang) [inline test/OpenTelemetry.Api.Tests/OpenTelemetry.Api.Tests.csproj:25]: These seem to be left over from previous restructuring.

**Review by utpilla — APPROVED** [review]
Thanks! Looks much cleaner now.

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**